rotkehlchen/data_import/importers/blockfi_trades.py

```diff
@@ -0,0 +1,91 @@
+from rotkehlchen.types import Location, Price, TradeType
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BlockfiTradesImporter(BaseExchangeImporter):
+    def _consume_blockfi_trade(self, csv_row: Dict[str, Any], **kwargs: Any) -> None:
+        """
+        Consume the file containing only trades from BlockFi. As per my investigations
+        (@yabirgb) this file can only contain confirmed trades.
+        - UnknownAsset
+        - DeserializationError
+        """
+        formatstr = kwargs.get('timestamp_format')
+        timestamp = deserialize_timestamp_from_date(
+            date=csv_row['Date'],
+            formatstr=formatstr if formatstr is not None else '%Y-%m-%d %H:%M:%S',
+            location='BlockFi',
+        )
+
+        buy_asset = symbol_to_asset_or_token(csv_row['Buy Currency'])
+        buy_amount = deserialize_asset_amount(csv_row['Buy Quantity'])
+        sold_asset = symbol_to_asset_or_token(csv_row['Sold Currency'])
+        sold_amount = deserialize_asset_amount(csv_row['Sold Quantity'])
+        if sold_amount == ZERO:
+            log.debug(f'Ignoring BlockFi trade with sold_amount equal to zero. {csv_row}')
+            return
+        rate = Price(buy_amount / sold_amount)
+        trade = Trade(
+            timestamp=timestamp,
+            location=Location.BLOCKFI,
+            base_asset=sold_asset,
+            quote_asset=buy_asset,
+            trade_type=TradeType.SELL,
+            amount=sold_amount,
+            rate=rate,
+            fee=None,  # BlockFI doesn't provide this information
+            fee_currency=None,
+            link='',
+            notes=csv_row['Type'],
+        )
+        self.add_trade(trade)
+
+    def import_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, str]:
+        """
+        Information for the values that the columns can have has been obtained from
+        the issue in github #1674
+        """
+        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+            data = csv.DictReader(csvfile)
+            for idx, row in enumerate(data):
+                if idx % ROWS_PER_CONTEXT_SWITCH == 0:
+                    gevent.sleep(0.1)
+                try:
+                    self._consume_blockfi_trade(row, **kwargs)
+                except UnknownAsset as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'During BlockFi CSV import found action with unknown '
+                        f'asset {e.asset_name}. Ignoring entry',
+                    )
+                    continue
+                except DeserializationError as e:
+                    self.db.msg_aggregator.add_warning(
+                        f'Deserialization error during BlockFi CSV import. '
+                        f'{str(e)}. Ignoring entry',
+                    )
+                    continue
+                except KeyError as e:
+                    return False, str(e)
+        self.flush_all()
```

Comment:
Oh, I see what you did here.
Rather than repeating this method call everywhere. Why not make abstract it inside the `BaseExchangeImporter` class.

Reply:
The reason to repeat it everywhere actually is to be sure that everything is saved before the `return` statement. I got your idea about putting it into the abstract class but I liked this one more because it gives more control for importers. I mean, importers might want handling database errors in their own ways but if database access happens after an importer finishes, it looses the control.

Reply:
Yes, it does make sense.  👍
Lefteris might have a totally different approach than both our methods lol ;)